flink-tests/src/test/java/org/apache/flink/test/javaApiOperators/GroupReduceITCase.java

```diff
@@ -434,6 +431,95 @@ public void testCorrectnessOfGroupreduceWithDescendingGroupSort() throws Excepti
 	}
 
 	@Test
+	public void testCorrectnessOfGroupreduceWithExplicitPartitionOfReducer() throws Exception {
+		/*
+		 * check correctness of groupReduce with descending group sort
+		 */
+		final ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+		env.setParallelism(1);
```

Comment:
If we run this test with parallelism `1`, we cannot check if the partitioning is working correctly. Also use a custom `GroupReduceFunction` that implements the `GroupCombineFunction` interface. The combine function should be implemented in a way that we can actually verify that it was called.